crates/cranelift/src/debug/transform/mod.rs

```diff
@@ -124,3 +150,34 @@ pub fn transform_dwarf(
         strings: out_strings,
     })
 }
+
+fn replace_unit_from_split_dwarf<'a>(
+    unit: &'a Unit<gimli::EndianSlice<'a, gimli::LittleEndian>, usize>,
+    dwp: &DwarfPackage<gimli::EndianSlice<'a, gimli::LittleEndian>>,
+    parent: &Dwarf<gimli::EndianSlice<'a, gimli::LittleEndian>>,
+) -> Option<(
+    Unit<gimli::EndianSlice<'a, gimli::LittleEndian>, usize>,
+    Dwarf<gimli::EndianSlice<'a, gimli::LittleEndian>>,
+)> {
+    if let Some(dwo_id) = unit.dwo_id {
```

Comment:
Idiomatically I might recommend some use of `?` and early-return to avoid the "tower" here with lots of indentation. For example this line here can be replaced with:

```rust
let dwo_id = unit.dwo_id?;
```

(and no indentation needed afterwards).

The next line could be replaced with:

```rust
let cu = dwp.find_cu(dwo_id, parent).ok()??;
```

(the double `??` is a bit odd but it handles the two layers of `Result<Option<_>>`)

Reply:
right-o